app/Models/Course.php

```diff
@@ -18,14 +18,72 @@ class Course extends Model
         'description'
     ];
 
+    public function scopeSearch($query, $data)
+    {
+        if (isset($data['course_name'])) {
+            $query->where('name', 'LIKE', "%". $data['course_name'] ."%");
+        }
+
+        if (isset($data['course_create'])) {
+            $data['course_create'] == "newest" ? $query->orderby('courses.created_at', 'DESC') : $query->orderby('courses.created_at');
+        }
+
+        if (isset($data['teacher_id']) && !empty($data['teacher_id'])) {
+            $query->join('teacher_course', 'courses.id', '=', 'teacher_course.course_id')
+                ->where('user_id', $data['teacher_id']);
+        }
+
+        if (isset($data['order_by_learner']) && !empty($data['order_by_learner'])) {
+            $query->withCount('users');
+            $query->orderBy('users_count', $data['order_by_learner']);
+        }
+
+        if (isset($data['order_by_time']) && !empty($data['order_by_time'])) {
+            $query->withCount('lessons')->withSum('lessons', 'times');
+            $query->orderBy('lessons_sum_times', $data['order_by_time']);
+        }
+
+        if (isset($data['order_by_lesson']) && !empty($data['order_by_lesson'])) {
+            $query->withCount('lessons');
+            $query->orderBy('lessons_count', $data['order_by_lesson']);
+        }
+
+        if (isset($data['tags']) && count($data['tags']) > 0) {
+            $query->join('course_tag', 'courses.id', '=', 'course_tag.course_id')
```

Comment:
câu query này hiểu như thế nào nhỉ?

Reply:
> câu query này hiểu như thế nào nhỉ?

select * 
from `courses` inner join `course_tag` on `courses`.`id` = `course_tag`.`course_id` 
where `tag_id` in (1, 3, 2)

Reply:
@phamthimylinhhaui hiểu theo lời nói là như thế nào?

Reply:
> @phamthimylinhhaui hiểu theo lời nói là như thế nào?

nối bảng courses với bảng course_tag,
khi có tag_id thuộc một trong những input truyền vào 
thì trả về các bản ghi thỏa mãn đk gồm tất cả các trường trong bảng courses